### PR TITLE
feat(flag): --path filtering - refs #35

### LINKED ISSUE #35: feat: Add options for which directories to filter & group to
## Expected Behavior

```bash
git-chglog ... --directories foo/bar/baz a/b/c
```

would
* filter the git log to just the entries that happened within the given directories
* uniquify the resulting set of entries

(I don't mind about a changelog entry appearing in more than one group's worth)

I work within a monorepo (with bazel), so we want to generate changelogs for just particular parts when we release those particular parts. I'd want to use bazel query to find all directories that a thing-being-released plus all it depends on, then generate a changelog for that list.

## Actual Behavior

(Feature doesn't exist)

## Steps to Reproduce (including precondition)

(Feature doesn't exist)

## Screenshot on This Problem (if possible)

(Feature doesn't exist)

## Your Environment

(Feature doesn't exist)

### PR DESCRIPTION
**Pull request by frimik**
## What does this do / why do we need it?

Filters by `--path`

## What should your reviewer look out for in this PR?

Not sure... I think this may not be mergeable. I made it very very quickly.

## Check lists

* [x] Test passed
* [x] Coding style (indentation, etc)

## Which issue(s) does this PR fix?

fixes #35

### CODE REVIEW

**Comment by frimik** [comment]
```sh
	git-chglog .
```

I use it like this... seems to work...

```Makefile
.PHONY: changelog
changelog: NEXT ?=
changelog: .chglog/config.yml
changelog: ## Generate Changelog
	@ $(MAKE) --no-print-directory log-$@
	cd $(GIT_ROOT) && git-chglog --tag-filter-pattern v[0-9]+.[0-9]+.[0-9]+$$ --output CHANGELOG.md $(NEXT)
	@ git add $(GIT_ROOT)/CHANGELOG.md
	@ git commit -m "Update Changelog" $(GIT_ROOT)/CHANGELOG.md
	@ echo "I've now committed a new CHANGELOG.md. It's up to you to push it."

.PHONY: changelog
changelog-local: NEXT ?=
changelog-local: .chglog/config.yml
changelog-local: ## Generate Changelog in the current directory
	@ $(MAKE) --no-print-directory log-$@
	git-chglog --config $(GIT_ROOT)/.chglog/config.yml --tag-filter-pattern v[0-9]+.[0-9]+.[0-9]+$$ --path . --output CHANGELOG.md $(NEXT)
	@ git add CHANGELOG.md
	@ git commit -m "Update Changelog"
	@ echo "I've now committed a new CHANGELOG.md. It's up to you to push it."

```

**Comment by wadackel** [comment]
Hi @frimik, Thank you for contributing! And I'm sorry too late... :sob:

>  I think this may not be mergeable.

Is there anything specifically that worries you :dog: ?

**Comment by frimik** [comment]
@wadackel was mainly thinking code style and lack of tests. I use it and it's working good so far :).

**Comment by ghostsquad** [comment]
@frimik can you rebase this please? if you need some help on tests, I'm happy to try to contribute.

File: cmd/git-chglog/main.go [hunk 1]
```diff
@@ -76,6 +80,12 @@ func CreateApp(actionFunc cli.ActionFunc) *cli.App {
 			Usage: "generate the git-chglog configuration file in interactive",
 		},
 
+		// path
+		cli.StringSliceFlag{
```
Comment (clok) [inline cmd/git-chglog/main.go:84]:
```suggestion
		&cli.StringSliceFlag{
```

**Review by clok — CHANGES_REQUESTED** [review]
@frimik Can you please rebase from `master`? We have updated `urfave/cli` to `v2` and this will require a minor change in your PR. 

```
cmd/git-chglog/main.go:84:22: cannot use cli.StringSliceFlag literal (type cli.StringSliceFlag) as type cli.Flag in slice literal:
	cli.StringSliceFlag does not implement cli.Flag (Apply method has pointer receiver)
```

I noted the change inline. I tested locally with the rebase and change, all looks good! Thank you for the contribution.

File: cmd/git-chglog/main.go (hunk 1 above)
Comment (frimik) [reply cmd/git-chglog/main.go]: Hopefully new commits resolves this concern.

Comment (clok) [reply cmd/git-chglog/main.go]: Thanks for pushing the updates. Looks good!

**Review by clok — APPROVED** [review]
Nice work. Tested locally as well. I can see this be useful.

**Comment by clok** [comment]
@wadackel got a Moment to take a look at the updated code?

**Comment by mavogel** [comment]
@frimik could you rebase :) Then let's merge this useful feature

**Comment by frimik** [comment]
> @frimik could you rebase :) Then let's merge this useful feature

Freshly rebased on top of the new Jira stuff!

**Comment by clok** [comment]
@mavogel could you take a look at the hanging check? I think it may be a legacy check again.

**Pull request merged**